File: python/where-my-anagrams.py
# Where my anagrams at? 
# write a function that will find all the anagrams of a word from a list. two inputs - a word and an array of words. return an array of all the anagrams in the list or an empty array if there are none
from collections import Counter
def anagrams(word, words):
    counter = Counter(word)
    anagrams = []
    print(counter)
    for w in words:
        if len(w) == len(word):
            c = Counter(w)
            if len(counter-c) == 0:
                anagrams.append(w)
    print(anagrams)        
            

# anagrams('abba', ['aabb', 'abcd', 'bbaa', 'dada'])
# anagrams('racer', ['crazer', 'carer', 'racar', 'caers', 'racer'])

File: javascript/where-my-anagrams.js
// Where my anagrams at? 5kyu
// Write a function that will find all the anagrams of a word from a list. You will be given two inputs a word and an array with words. You should return an array of all the anagrams or an empty array if there are none.


function anagrams(word, words) {
    let anagrams = []
    let newWords = words.filter(w => w.length === word.length)
    let original = word.split('').sort().join('')
    for (let w of words) {
        let temp = w.split('').sort().join('')
        if (original === temp) {
            anagrams.push(w)
        }
    }
    console.log(anagrams)
    return anagrams
}


// anagrams('abba', ['aabb', 'abcd', 'bbaa', 'dada'])
// anagrams('racer', ['crazer', 'carer', 'racar', 'caers', 'racer'])
// anagrams('laser', ['lazing', 'lazy', 'lacer'])
